ext4: don't set extents flag for _any_ symlinks
From: Eric Sandeen <sandeen@redhat.com>

As symlinks are limited to a single block anyway, and e2fsck
doesn't expect to find it set, don't set the extents flag on 
any type of symlinks at all: fast/in-inode, or the 
external-block flavor.

There are a lot of filesystems out there by now w/ exent-style 
symlink blocks though, so e2fsck should probably be able to 
repair that at some point...

Signed-off-by: Eric Sandeen <sandeen@redhat.com>

---
 fs/ext4/ialloc.c |    2 +-
 fs/ext4/namei.c  |    1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

Index: linux-2.6.25-rc2/fs/ext4/namei.c
===================================================================
--- linux-2.6.25-rc2.orig/fs/ext4/namei.c	2008-02-18 15:02:47.000000000 -0800
+++ linux-2.6.25-rc2/fs/ext4/namei.c	2008-02-18 21:18:27.000000000 -0800
@@ -2220,7 +2220,6 @@ retry:
 		inode->i_op = &ext4_fast_symlink_inode_operations;
 		memcpy((char*)&EXT4_I(inode)->i_data,symname,l);
 		inode->i_size = l-1;
-		EXT4_I(inode)->i_flags &= ~EXT4_EXTENTS_FL;
 	}
 	EXT4_I(inode)->i_disksize = inode->i_size;
 	err = ext4_add_nondir(handle, dentry, inode);
Index: linux-2.6.25-rc2/fs/ext4/ialloc.c
===================================================================
--- linux-2.6.25-rc2.orig/fs/ext4/ialloc.c	2008-02-18 21:17:53.000000000 -0800
+++ linux-2.6.25-rc2/fs/ext4/ialloc.c	2008-02-18 21:18:27.000000000 -0800
@@ -744,7 +744,7 @@ got:
 		ext4_std_error(sb, err);
 		goto fail_free_drop;
 	}
-	if (test_opt(sb, EXTENTS)) {
+	if (test_opt(sb, EXTENTS) && !S_ISLNK(mode)) {
 		EXT4_I(inode)->i_flags |= EXT4_EXTENTS_FL;
 		ext4_ext_tree_init(handle, inode);
 		err = ext4_update_incompat_feature(handle, sb,
